### PR TITLE
Consolidate RecipeSelector into SimplePlanner and move to planning

### PR DESCRIPTION
**Pull request by raulverag**
Checks that the modality from the arc is compatible with the selected recipe.
Allows for duplicate triggers, as different recipes might be activated by the same trigger but a different modality.

### CODE REVIEW

File: src/planning/simple-planner.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    return true;
+  }
+}
+
+/**
+ * A very simple planner that looks up recipes based on triggers in the
+ * manifest file matching requests.
+ */
+export class SimplePlanner {
+  private _table: Match[] = [];
```
Comment (mmandlis) [inline src/planning/simple-planner.ts]: nit - `_table` is not a very descriptive name. `matches` maybe? or just `recipeTriggers`? also (nit-nit) you could make it a `public final` and then no getter would be needed.

File: src/planning/simple-planner.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      });
+    });
+  }
+
+  // Returns a Recipe or null if no suitable recipe is found. A suitable recipe
+  // is the first recipe in the table that meets three conditions:
+  // 1 - All pairs in the trigger are in the request.
+  // 2 - The modality of the arc is compatible with the recipe.
+  // 3 - The recipe resolves.
+  // TODO:
```
Comment (mmandlis) [inline src/planning/simple-planner.ts]: nit - "TODO: " is empty

File: src/planning/simple-planner.ts (hunk 1 above)
Comment (raulverag) [reply src/planning/simple-planner.ts]: Renamed it to `_recipesByTrigger`. `public final` didn't work, causing the TS compiler to complain. It didn't seem worthwhile to chase it down.

File: src/planning/simple-planner.ts (hunk 2 above)
Comment (raulverag) [reply src/planning/simple-planner.ts]: Deleted. Thanks.

**Pull request merged**